diff --git a/node_modules/r3f6/dist/react-three-fiber.esm.js b/node_modules/r3f6/dist/react-three-fiber.esm.js
index 768ff87..deec684 100644
--- a/node_modules/r3f6/dist/react-three-fiber.esm.js
+++ b/node_modules/r3f6/dist/react-three-fiber.esm.js
@@ -1258,7 +1258,7 @@ const createStore = (applyProps, invalidate, advance, props) => {
 
 
     gl.setPixelRatio(viewport.dpr);
-    gl.setSize(size.width, size.height);
+    gl.setSize(size.width, size.height, false);
   }, state => [state.viewport.dpr, state.size], shallow);
   const state = rootState.getState(); // Update size
 
@@ -1778,4 +1778,4 @@ reconciler.injectIntoDevTools({
   version: '17.0.2'
 });
 
-export { Canvas, threeTypes as ReactThreeFiber, roots as _roots, act, addAfterEffect, addEffect, addTail, advance, applyProps, context, createPortal, dispose, createPointerEvents as events, extend, invalidate, reconciler, render, unmountComponentAtNode, useFrame, useGraph, useLoader, useThree };
+export { Canvas, threeTypes as ReactThreeFiber, roots as _roots, act, addAfterEffect, addEffect, addTail, advance, applyProps, context, createPortal, dispose, createPointerEvents as events, createEvents, extend, invalidate, reconciler, render, unmountComponentAtNode, useFrame, useGraph, useLoader, useThree };
